File: packages/sdkwork-utils-java/src/main/java/com/sdkwork/utils/IdUtils.java
package com.sdkwork.utils;

import java.security.SecureRandom;
import java.util.UUID;

public final class IdUtils {
    private static final SecureRandom RANDOM = new SecureRandom();
    private static final String ALPHANUMERIC =
            "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

    private IdUtils() {
    }

    public static String uuid() {
        return UUID.randomUUID().toString();
    }

    public static String randomString(int length) {
        StringBuilder builder = new StringBuilder(length);
        for (int index = 0; index < length; index++) {
            builder.append(ALPHANUMERIC.charAt(RANDOM.nextInt(ALPHANUMERIC.length())));
        }
        return builder.toString();
    }
}

File: packages/sdkwork-utils-kotlin/src/main/kotlin/com/sdkwork/utils/IdUtils.kt
package com.sdkwork.utils

import java.security.SecureRandom
import java.util.UUID

object IdUtils {
    private val RANDOM = SecureRandom()
    private const val ALPHANUMERIC =
        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"

    fun uuid(): String {
        return UUID.randomUUID().toString()
    }

    fun randomString(length: Int): String {
        val builder = StringBuilder(length)
        for (index in 0 until length) {
            builder.append(ALPHANUMERIC[RANDOM.nextInt(ALPHANUMERIC.length)])
        }
        return builder.toString()
    }
}
